src/http.c

```diff
@@ -621,6 +621,44 @@ static int try_otp_auth(struct tunnel *tunnel, const char *buffer,
 }
 
 
+/*
+ * Authenticates to gateway by sending username and password.
+ *
+ * @return  1   in case of success
+ *          < 0 in case of error
+ */
+int saml_login(struct tunnel *tunnel)
+{
+	log_debug("SAML login\n");
+
+	int ret;
+	ssl_connect(tunnel);
+
+	char uri[1024];
+	snprintf(uri, sizeof(uri), "/remote/saml/auth_id?id=%s", tunnel->config->saml_session_id);
```

Comment:
In case the session ID gets very long, the resulting string will not be null terminated anymore.
This lead to other parts dealing with this string behave wrong.
There should either be a check if the string fits, or at least force it to be null terminated.

Reply:
`snprintf()` will truncate and null-terminate if it's too long. 
An undetected truncated string might be an issue, so better check the return value anyway.